python/paddle/fluid/parallel_executor.py

```diff
@@ -96,19 +96,31 @@ def __init__(self,
 
         if build_strategy is None:
             build_strategy = BuildStrategy()
-        build_strategy.num_trainers = num_trainers
-        build_strategy.trainer_id = trainer_id
```

Comment:
The `trainer_id` and `num_trainers` of `build_strategy` should not be overwritten by the passed `trainer_id` and `num_trainers` of `ParallelExecutor`.